reasoner_pydantic/shared.py

```diff
@@ -10,15 +13,44 @@
 from .base_model import BaseModel
 from .utils import HashableSequence
 
-
+# TODO: potential add validation for structure of CURIE
 class CURIE(str):
     """Compact URI."""
 
 
+class KnowledgeType(str, Enum):
+    "Knowledge Type."
+
+    lookup = "lookup"
+    inferred = "inferred"
+
+
 class EdgeIdentifier(str):
     """Identifier for an edge in a knowledge graph"""
 
 
+class RecurrsiveAttribute(BaseModel):
```

Comment:
Spelling - RecursiveAttribute (one r in the middle)

Reply:
Fixed